### PR TITLE
Add support for `console.time`

### PR DESCRIPTION
**Pull request by legendecas**

### CODE REVIEW

**Comment by yorkie** [comment]
@legendecas Awesome, could you please add some test case for your code, see https://github.com/Rokid/ShadowNode/tree/master/test/run_pass :)

File: src/js/iotjs.js [hunk 1]
```diff
@@ -222,6 +222,32 @@
...
 
+  var _hrtime = process.hrtime;
+  var NANOSECOND_PER_SECONDS = 1e9;
+  process.hrtime = function hrtime (time) {
+    var curr = _hrtime();
+    if (time == null) {
+      return curr;
+    }
+    if (!Array.isArray(time)) {
+      var error = TypeError('[ERR_INVALID_ARG_TYPE]: The "time" argument must be of type Array. Received type ' + typeof time);
```
Comment (yorkie) [inline src/js/iotjs.js]: missing the `new` keyword :)

File: src/js/iotjs.js [hunk 2]
```diff
@@ -222,6 +222,32 @@
...
+      error.code = 'ERR_INVALID_ARG_TYPE';
+      throw error;
+    }
+    if (time.length != 2) {
+      var error = TypeError('[ERR_INVALID_ARRAY_LENGTH]: The array "time" (length ' + String(time.length) + ') must be of length 2.');
```
Comment (yorkie) [inline src/js/iotjs.js]: Ditto, and the line should be in 80 characters :)

File: src/js/console.js [hunk 3]
```diff
@@ -18,7 +18,23 @@ var util = require('util');
 
 
 function Console() {
-  // Empty
+  var prop = {
+    writable: true,
+    enumerable: false,
+    configurable: true
+  };
+  prop.value = stdout;
```
Comment (yorkie) [inline src/js/console.js]: Don't use the same property object for different defines, use `Object.assign()` to generate separate property :)

File: src/js/console.js [hunk 4]
```diff
@@ -40,15 +56,25 @@ Console.prototype.error = function() {
...
-  log: console.log.bind(console),
-  debug: console.debug.bind(console),
-  info: console.info.bind(console),
-  warn: console.warn.bind(console),
-  error: console.error.bind(console),
-  Console: Console,
-  _stdout: stdout,
-  _stderr: stderr,
+  var time = this._times[label];
+  if (time == null) {
```
Comment (yorkie) [inline src/js/console.js]: strict equal?

Comment (legendecas) [reply src/js/console.js]: `variable == null` is both true if variable is `null` or `undefined`. I think this shall be one exception for rule of strict equal.

Comment (yorkie) [reply src/js/console.js]: See https://github.com/airbnb/javascript#comparison--eqeqeq, we should follow this lint rule.

**Comment by yorkie** [comment]
Rad @legendecas !

**Pull request merged**